src/rateLimiters/slidingWindowCounter.ts

```diff
@@ -112,10 +112,22 @@ class SlidingWindowCounter implements RateLimiter {
         };
 
         // if request time is in a new window
-        if (timestamp >= window.fixedWindowStart + this.windowSize + 1) {
-            updatedUserWindow.previousTokens = updatedUserWindow.currentTokens;
-            updatedUserWindow.currentTokens = 0;
-            updatedUserWindow.fixedWindowStart = window.fixedWindowStart + this.windowSize + 1;
+        if (timestamp >= window.fixedWindowStart + this.windowSize) {
+            // if more than one window was skipped
+            if (timestamp >= window.fixedWindowStart + this.windowSize * 2) {
+                // calculates how many windows may have been skipped since last request
```

Comment:
I'm fine with this approach. We could simplify this by starting the new window at the time of the current request.

Reply:
Good idea. Will update